include/turtle/constraint.hpp

```diff
@@ -136,7 +135,7 @@ namespace detail
             template< typename Actual > \
             bool operator()( const Actual& actual ) const \
             { \
-                return Expr; \
+                (void) actual; return Expr; \
```

Comment:
We could also remove the argument
```
bool operator()( const Actual& ) const \
```

Reply:
But Expr could want to use actual. And it often does. It is only sometimes that the argument is unused which creates the warning